bgpd/bgp_network.c

```diff
@@ -442,10 +448,10 @@ static int bgp_bind(struct peer *peer)
 	int myerrno;
 	char *name = NULL;
 
-	/* If not bound to an interface or part of a VRF, we don't care. */
-	if (!peer->bgp->vrf_id && !peer->ifname && !peer->conf_if)
+	/* If not bound to an interface or part of a VRF lite, we don't care. */
+	if (!(peer->bgp != NULL && vrf_is_mapped_on_netns(peer->bgp->vrf)) &&
+	    !peer->ifname && !peer->conf_if)
```

Comment:
about peer->bgp , using scan-build make, I encountered this issue.
do you see any solution to avoid that scan-build ignores that issue ?

Reply:
the assumption  It works because vrf_is_mapped_on_netns() checks if vrf->vrf_id == NS_DEFAULT is false.
because NS_DEFAULT and VRF_DEFAULt overlap, you need one more information to know if a vrf has a netns backend orn ot.

Reply:
please review the change

Reply:
Looks better now, thanks :)